src/module/rules/rule-element/lose-hit-points.ts

```diff
@@ -20,6 +25,27 @@ export class LoseHitPointsRuleElement extends RuleElementPF2e {
             actorUpdates["data.attributes.hp.value"] = Math.max(currentHP - value, 0);
         }
     }
+
+    override async preUpdate(changed: DeepPartial<ItemSourcePF2e>): Promise<void> {
+        if (!this.reevaluateOnUpdate || this.ignored) return;
+        const previousValue = Math.abs(Number(this.resolveValue()) || 0);
+        mergeObject(this.item.data, changed);
+        const newValue = Math.abs(Number(this.resolveValue()) || 0);
```

Comment:
Instead of mutating the current item, clone the source (`this.item.toObject()`), merge that, and fish out the same RE.

Reply:
Problem is I can't resolve the value of the new RE with the context of the preUpdated item since it doesn't exist.

Reply:
You can clone the item with the changes.

Reply:
Before I commit, I'd be doing something like:
```
const newItem = this.item.clone();
mergeObject(newItem.data, changed);
const rule = newItem.data.data.rules.find((r) => r.key === this.key);
const newValue = Math.abs(Number(this.resolveValue(rule?.value, 0, { resolvables: { item: newItem } })));
```
?

Reply:
This works I just want to make sure it's what you had in mind

Reply:
include the change in the clone method

Reply:
Nice, thank you, that's all in now